### PR TITLE
bpo-37499: Test various C calling conventions

### PR DESCRIPTION
**Pull request by encukou**
Add functions with various calling conventions to `_testcapi`, expose them as module-level functions, bound methods, class methods, and static methods, and test calling them and introspecting them through GDB.

https://bugs.python.org/issue37499

Co-authored-by: Jeroen Demeyer <J.Demeyer@UGent.be>
Automerge-Triggered-By: @pganssle

### CODE REVIEW

**Comment by encukou** [comment]
I started this on a train, and never got to finishing it up. Unfortunately it duplicates work in GH-14795.

File: Lib/test/test_call.py [hunk 1]
```diff
@@ -289,6 +170,187 @@ def test_oldargs1_2_kw(self):
...
+
+    def test_fastcall_keywords_ext(self):
+        self.assertEqual(
+            self.obj.meth_fastcall_keywords(*(1, 2), **{'a': 3, 'b': 4}),
+            (self.expected_self, (1, 2), {'a': 3, 'b': 4})
+        )
+
+
+@cpython_only
+class TestCallingConventionsFunction(TestCallingConventions):
```
Comment (pganssle) [inline Lib/test/test_call.py]: Unnecessary?

File: Lib/test/test_call.py [hunk 2]
```diff
@@ -289,6 +170,187 @@ def test_oldargs1_2_kw(self):
...
+    """Test calling module-level functions using various calling conventions"""
+
+
+
+@cpython_only
```
Comment (pganssle) [inline Lib/test/test_call.py]: May not need this, PyPy has `cpyext` and supports the C API.

**Review by pganssle — CHANGES_REQUESTED** [review]

**Comment by encukou** [comment]
I have made the requested changes; please review again.

**Comment by encukou** [comment]
The `Function "meth_noargs" not defined.` warnings on stderr aren't easy to get rid of (without masking other warnings by redirecting stderr): See the [comment in get_stack_trace](https://github.com/python/cpython/blob/master/Lib/test/test_gdb.py#L156)

**Review by pganssle — APPROVED** [review]
This looks good to me, I think we can go ahead and merge.

@encukou Let me know if you have any problems with the changes I pushed in d8a28ab29e668e5d66824fdaec5bc59cf914e6a7.

Also, I had to rebase this against master to resolve a merge conflict, so you'll need to do a force pull or something equivalent if you want to push more commits.

File: Lib/test/test_gdb.py [hunk 3]
```diff
@@ -838,47 +838,66 @@ def test_gc(self):
...
+                '_testcapi.MethClass',
+                '_testcapi.MethClass()',
+                '_testcapi.MethStatic()',
+
+                # XXX: bound methods don't yet give nice tracebacks
+                # '_testcapi.MethInstance()',
+            ):
+                with self.subTest(f'{obj}.{func_name}'):
+                    cmd = textwrap.dedent(f'''
+                        import _testcapi  # (not always needed)
```
Comment (encukou) [inline Lib/test/test_gdb.py]:
```suggestion
                        import _testcapi
```

**Pull request merged**